builder_pattern/builder.go

```diff
@@ -0,0 +1,11 @@
+package builder_pattern
+
+type builder interface {
```

Comment:
```suggestion
type houseBuilder interface {
```

Reply:
let's add doc that houseBuilder contains all the methods that are required to create a house